### PR TITLE
Add helpers for paused clusters

### PR DESCRIPTION
**Pull request by randomvariable**
A speculative PR to make some of the checks around paused clusters or resources reusable across providers.

### CODE REVIEW

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -107,6 +107,7 @@ func (r *KubeadmConfigReconciler) SetupWithManager(mgr ctrl.Manager, option cont
 				ToRequests: handler.ToRequestsFunc(r.ClusterToKubeadmConfigs),
 			},
 		).
+		WithEventFilter(util.external.FilterPausedAnnotations).
```
Comment (vincepri) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller.go]:
```suggestion
		WithEventFilter(external.FilterPausedAnnotations).
```

Autocomplete strikes back? :P

**Comment by vincepri** [comment]
Nice, this is a great improvement! 🎉 

what do you think about updating the v1alpha2 to v1alpha3 document and suggest to add this as well to providers' controllers?

**Comment by randomvariable** [comment]
Sure.

Will be out for rest of the day, then will get it working and solve the gopls horrors, probably by ditching VS Code once and for all.

**Comment by vincepri** [comment]
/milestone v0.3.x

Moving this to v0.3.x for now, given that the release is only next week

**Comment by randomvariable** [comment]
I'll convert this to an issue and close I think.

Already found some logic errors that have crept into CAPV. Will need to add some proper tests before making library functions.

**Pull request closed without merging**